SConstruct

```diff
@@ -7,14 +7,18 @@ import source.lib.JMSLab as jms
 sys.path.append('config')
 sys.dont_write_bytecode = True # Don't write .pyc files
 
-env = Environment(ENV = {'PATH' : os.environ['PATH']},
+vars = Variables()
+vars.Add('HANDOUT_SFIX', 'Set to string to create handout', '')
```

Comment:
@jmshapir I see! I think I can move the operation from `Sconstruct` to `Sconscript` with no problem. Beyond that, I am not sure, but I can investigate some more. Just to be clear, what you want is that in the `env.Lyx(target, source, <explicit option>)` call, the `<explicit option>` part will create a previously undefined variable?

Reply:
@ew487 thanks!

> Just to be clear, what you want is that in the `env.Lyx(target, source, <explicit option>)` call, the `<explicit option>` part will create a previously undefined variable?

The functionality you describe above sounds like it would work well.

If that's not possible, an alternative would be that the variable is always created by default "inside" the builder (and maybe set to null by default) so that the user can (optionally) specify it in the `env.LyX` call.

Since I'm envisioning most use cases won't require declaration of the suffix, I'm inclined to keep the suffix option as "quiet" as possible from the standpoint of the scons calls.

Reply:
@jmshapir Got it! Thanks for clarifying!

Reply:
@jmshapir This should be fixed now in [7990e84](https://github.com/JMSLab/Template/pull/62/commits/7990e842128b2e8c7a3dba6d2b1c43816a9f6c61).